src/argilla/client/feedback/dataset/local/base.py

```diff
@@ -0,0 +1,408 @@
+        """
+        if isinstance(records, (dict, FeedbackRecord)):
+            records = [records]
+
+        if len(records) == 0:
+            raise ValueError("Expected `records` to be a non-empty list of `dict` or `FeedbackRecord`.")
+
+        new_records = []
+        for record in records:
+            if isinstance(record, dict):
+                new_records.append(FeedbackRecord(**record))
+            elif isinstance(record, FeedbackRecord):
+                new_records.append(record)
+            else:
+                raise ValueError(
+                    "Expected `records` to be a list of `dict` or `FeedbackRecord`,"
+                    f" got type `{type(record)}` instead."
+                )
+        return new_records
+
+    def _validate_records(self, records: List[FeedbackRecord]) -> None:
+        """Validates the records against the schema defined by the `fields`.
+
+        Args:
+            records: a list of `FeedbackRecord` objects to validate.
+
+        Raises:
+            ValueError: if the `fields` schema does not match the `FeedbackRecord.fields` schema.
+        """
+        if self._fields_schema is None:
+            self._fields_schema = generate_pydantic_schema(self.fields)
+
+        for record in records:
+            try:
+                self._fields_schema.parse_obj(record.fields)
+            except ValidationError as e:
+                raise ValueError(
+                    f"`FeedbackRecord.fields` does not match the expected schema, with exception: {e}"
+                ) from e
+
+    def _parse_and_validate_records(
+        self,
+        records: Union[FeedbackRecord, Dict[str, Any], List[Union[FeedbackRecord, Dict[str, Any]]]],
+    ) -> List[FeedbackRecord]:
+        """Convenient method for calling `_parse_records` and `_validate_records` in sequence."""
+        records = self._parse_records(records)
+        self._validate_records(records)
+        return records
+
+    @requires_version("datasets")
+    def format_as(self, format: Literal["datasets"]) -> "Dataset":
+        """Formats the `FeedbackDataset` as a `datasets.Dataset` object.
+
+        Args:
+            format: the format to use to format the `FeedbackDataset`. Currently supported formats are:
+                `datasets`.
+
+        Returns:
+            The `FeedbackDataset.records` formatted as a `datasets.Dataset` object.
+
+        Raises:
+            ValueError: if the provided format is not supported.
+
+        Examples:
+            >>> import argilla as rg
+            >>> rg.init(...)
+            >>> dataset = rg.FeedbackDataset.from_argilla(name="my-dataset")
+            >>> huggingface_dataset = dataset.format_as("datasets")
+        """
+        if format == "datasets":
+            return self._huggingface_format(self)
+        raise ValueError(f"Unsupported format '{format}'.")
+
+    # TODO(davidberenstein1957): detatch unification into a mixin
+    def unify_responses(
```

Comment:
Maybe it's a nice time to detach this also from the current class and move it to a `UnificationMixin` or something similar?

Reply:
I agree with this.